### ISSUE TITLE
Can't apply label with empty values not applied on nodes

### ISSUE THREAD

**Issue by schabrolles** [open-cluster-management-io/policy-collection#425]
HI all,

I'm trying to create a policy to install automatically ODF on an openshift cluster.
For that, I need to label some nodes with `cluster.ocs.openshift.io/openshift-storage: ""`

My policy says: 
> notification - nodes [worker1.pink.edu.ihost.com] found as specified, therefore this Object template is compliant

But the label is not there :/ 

Here is the ComplicancePolicy used:

```
 apiVersion: policy.open-cluster-management.io/v1
  kind: ConfigurationPolicy
  metadata:
    name: odf-label-nodes
  spec:
    object-templates:
      - complianceType: musthave
        objectDefinition:
          apiVersion: v1
          kind: Node
          metadata:
            name: worker1.pink.edu.ihost.com
            labels:
              cluster.ocs.openshift.io/openshift-storage: ""
```

I've seen some other policies in this repo that are doing the same thing; but it doesn't seem to work. 
[community/CM-Configuration-Management/policy-label-worker-nodes.yaml](https://github.com/open-cluster-management-io/policy-collection/blob/main/community/CM-Configuration-Management/policy-label-worker-nodes.yaml)

I'm using openshift 4.13 an ACM 2.8.0
Any Idea ?

**Comment by gparvin** [comment]
I'll give it a try.  What I do is create a MachineSet with a Policy using the PolicyGenerator to setup some things before installing ODF as part of OPP.
- Create a MachineSet with the correct label in this PolicySet: https://github.com/open-cluster-management-io/policy-collection/tree/main/policygenerator/policy-sets/community/openshift-plus-setup
- Then I install ODF and the rest of OPP with this generator PolicySet: https://github.com/open-cluster-management-io/policy-collection/tree/main/policygenerator/policy-sets/stable/openshift-plus

**Comment by schabrolles** [comment]
Well, I'm using UPI installation mode for on premise cluster on ppc64le CPU arch servers... So for the moment, machineSet is not an option.

I notice it does not apply the label with empty value when the label does not exist.
But if the label is set with another value, the policy works and it can set the label to the "" value.... (very strange)

So I made it work with the following ugly workaround :/ 

```
    - objectDefinition:
        apiVersion: policy.open-cluster-management.io/v1
        kind: ConfigurationPolicy
        metadata:
          name: odf-label-nodes
        spec:
          pruneObjectBehavior: None
          remediationAction: inform
          severity: high
          object-templates-raw: |
            {{- range $item, $value := (lookup "v1" "Node" "default" "").items}}
              {{- if lt $item 3}}
             - complianceType: musthave
               objectDefinition:
                 apiVersion: v1
                 kind: Node
                 metadata:
                   name: {{ .metadata.name }}
                   labels:
                   {{ $label := index .metadata.labels "cluster.ocs.openshift.io/openshift-storage"}}
                   {{- if $label }}
                      cluster.ocs.openshift.io/openshift-storage: ''
                   {{- else}}
                    {{- if ne $label "" }}
                      cluster.ocs.openshift.io/openshift-storage: none
                     {{- end }}
                   {{- end }}
              {{- end }}
             {{- end }}
```

**Comment by gparvin** [comment]
Yes, that's what I just noticed.  You can add any "extra" label or annotation with a value set and then you will get your labels with "" values.

**Comment by schabrolles** [comment]
This is a bit complicated and seems to be a bug. 🤔

**Comment by gparvin** [comment]
I did create a bug for this. I was thinking it would automatically link to this issue
- https://issues.redhat.com/browse/ACM-6760

**Comment by gparvin** [comment]
I found out a bug already existed for this: https://issues.redhat.com/browse/ACM-5398
That bug is being worked on now (this sprint).

**Comment by gparvin** [comment]
The fix was delivered in ACM 2.8.2.  I think it's been available for about a month but I forgot to follow up here.  Let me know if we can close this issue.

**Comment by gparvin** [comment]
Please re-open this issue if you find that the problem is not fixed in ACM 2.8 or 2.9.  Thanks
